src/main/java/Duke.java

```diff
@@ -52,16 +76,16 @@ public static void runUI(ArrayList<Task> tasks){
 
                 case "deadline":
                     try {
-                        taskName = taskName.substring(9);
+                        taskName = taskName.trim().substring(9);
                         date = taskName.split("/");
                         task = new Deadline (date[0],date[1].substring(3));
                         addTaskIn(task,tasks);
-                    } catch (StringIndexOutOfBoundsException ex2){
+                    } catch (StringIndexOutOfBoundsException exDeadline1){
                         System.out.println("    ____________________________________________________________");
                         System.out.println("     ☹ OOPS!!! The description of a deadline cannot be empty.");
                         System.out.println("    ____________________________________________________________");
 
-                    } catch (IndexOutOfBoundsException ex3){
+                    } catch (IndexOutOfBoundsException exDeadline2){
```

Comment:
I like how you handled the different missing arguments. You could rename the exception variable name to be more meaningful, or simply use `e`/`ex`.